backend/tests/test_oauth.py

```
@@ -1576,6 +1576,65 @@ def test_callback_unknown_state_redirects_with_error():
+        assert "evil.example" not in r.headers["location"]
+
+
+def test_oauth_failure_reasons_log_at_warning(caplog):
+    # Regression: these lines are the ONLY record of why a sign-in failed, and under
+    # uvicorn's default logging config the root logger has no handler while app.* sits
+    # at effective level WARNING — so an INFO record is discarded and the operator is
+    # left with a toast and nothing to debug. They must be WARNING or above.
+    import logging
+
+    with TestClient(app) as c:
+        with caplog.at_level(logging.WARNING, logger="app.api.auth"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)